ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/CreateTableEvent.java

```diff
@@ -63,10 +63,19 @@ private List<HivePrivilegeObject> getInputHObjs() {
     PreCreateTableEvent       event = (PreCreateTableEvent) preEventContext;
     Table                     table = event.getTable();
     String                    uri   = getSdLocation(table.getSd());
+    // if the table is storagehandler based table, need storageuri as hive privilege objects
+    //  storagehandler type + cluster + location
+    String storagehandler_uri = table.getParameters().getOrDefault("storage_handler", null);
+    if (storagehandler_uri != null && storagehandler_uri.contains("iceberg")){
+      storagehandler_uri = "hiveicebergstoragehandler://"+ table.getParameters().getOrDefault("metadata_location", "");
```

Comment:
Storage_uri formation like this is not recommended. Please use this method https://github.com/apache/hive/blob/master/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java#L1065-L1093 to obtain URI. This is consistent with what we do in HS2: https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/security/authorization/command/CommandAuthorizerV2.java#L208
For reference you could do something like this: https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/AlterTableEvent.java#L108-L112